sentry_sdk/integrations/arq.py

```diff
@@ -165,11 +160,10 @@ def _wrap_coroutine(name, coroutine):
     # type: (str, WorkerCoroutine) -> WorkerCoroutine
     async def _sentry_coroutine(ctx, *args, **kwargs):
         # type: (Dict[Any, Any], *Any, **Any) -> Any
-        hub = Hub.current
-        if hub.get_integration(ArqIntegration) is None:
+        if sentry_sdk.get_client().get_integration(ArqIntegration) is None:
             return await coroutine(ctx, *args, **kwargs)
 
-        hub.scope.add_event_processor(
+        Scope.get_current_scope().add_event_processor(
```

Comment:
Are we sure this event processor belongs on the scope, and not the isolation scope?

Reply:
This I wasn't sure about. Thinking about it again, I think isolation scope makes more sense. Will change.

Reply:
There's also another `current_scope` a bit [up](https://github.com/getsentry/sentry-python/pull/2878/files#diff-85f717047dcb658107e9e812c993f2487fd94da0a9db77063bbe39e1cc783a09R134). I think this is what confused me. It feels like both should be isolation scopes. What do you think @szokeasaurusrex @antonpirker?

Reply:
I think event processors should be on the isolation scope, because they should be valid for the whole request.

Reply:
I think what @antonpirker said makes sense

Reply:
Tried it out now and changing either of them (or both) to work with the isolation scope makes the tests fail. 🙃 Investigating...

Reply:
Ok this works: https://github.com/getsentry/sentry-python/pull/2878/commits/9d197a3e6d801cabbcd233d97b5f0dba71420c14
* the current transaction is saved on the current scope, so all scopes in the arq integration that try to update transaction attrs also have to go to the current scope
* the event processor itself is now on the isolation scope